[Backport] [Nexus] Revert to single wv session

glennguy opened the pull request:
## Description
<!--- Provide a general summary of your change in the Pull Request title above -->
<!--- Describe your change in detail here. -->

Backport of #1165

This also include backports of:
#1235
#1244

## Motivation and context
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

## How has this been tested?
<!--- Please describe in detail how you tested your change. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->

## Screenshots (if appropriate):

## Types of change
<!--- What type of change does your code introduce? Put an `x` in all the boxes that apply like this: [X] -->
- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:
<!--- Go over all the following points, and put an `X` in all the boxes that apply like this: [X] -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the Wiki documentation
- [ ] I have updated the documentation accordingly

glennguy committed:
[Widevine] Remove multi subsample decrypting

note - decrypting multiple subsamples is possible but we transform the data first so the decrypter sees 1 subsample.

Decrypting with subsamples > 1 seems to be fully broken now, at least as far as I can tell.

glennguy committed:
[Widevine] [Android] Use single session where possible

Previously turned off to avoid certain pixelation issues. This should be fixed properly rather than this workaround.

glennguy committed:
[DRM] send crypto mode and pattern with fragment info

To facilitate CBCS decryption on a single session

glennguy committed:
[DRM] revert to using shared decrypters between stream types

Decrypting audio and video can use the same session if both keys are present.

glennguy committed:
Bump Bento4

glennguy committed:
Refactor some Bento4 added methods as subclassed code

CastagnaIT reviewed (commented):
seem needs to be synced with original pr

CastagnaIT committed:
[AdaptiveCencSampleDecrypter] Removed ss decrypter downcast

Looks like downcasting `AP4_CencSingleSampleDecrypter` to derived class `Adaptive_CencSingleSampleDecrypter` dont works on android, so when `DecryptSampleData` try to do the dynamic_cast it fails and dont decrypt content
since we can avoid the downcasting better do it and remove the dynamic_cast

glennguy committed:
[DASH] parse encryption scheme value

CastagnaIT reviewed (approved):
i havent tested locally but lgtm

glennguy commented:
Tested on CBCS and CENC streams, all looking good :)
